### PR TITLE
Reader position

### LINKED ISSUE #46: StringReader should maintain current position when reading

### PR DESCRIPTION
**Pull request by kpodsiad**
closes #46

### CODE REVIEW

File: yaml/shared/src/main/scala/org/virtuslab/yaml/internal/load/reader/Reader.scala [hunk 1]
```diff
@@ -1,37 +1,66 @@
...
 
   override def peekNext(): Option[Char] = peek(1)
   override def peekN(n: Int): String    = (0 until n).map(peek(_)).flatten.mkString("")
   override def isNextWhitespace         = peekNext().exists(_.isWhitespace)
+  override def isNewline                = peek().exists(c => c == '\n' || c == '\r')
+
+  private inline def nextLine() = { column = 1; line += 1 }
+  private def skipAndMantainPosition() =
+    val next = in.charAt(offset)
+    if next == '\r' then
```
Comment (lwronski) [inline yaml/shared/src/main/scala/org/virtuslab/yaml/internal/load/reader/Reader.scala:42]: The same as above

**Review by lwronski — COMMENTED** [review]
You should add more test, one test is not enough for this functionality which is critical to displaying detailed parsing errors.

**Comment by kpodsiad** [comment]
> You should add more test, one test is not enough for this functionality which is critical to displaying detailed parsing errors.

I agree, however, this functionality can be tested indirectly by parsing errors tests (after we implement both functionality and tests of course 😅). If you have particular test cases in your mind you are welcome to share them because, honestly, I struggle with potential new test scenarios.

**Review by lwronski — APPROVED** [review]
LGTM

**Pull request merged**